### ISSUE TITLE
Cmake issue with cuDNN; Docker container doesn't seem to have cuDNN?

### ISSUE THREAD

**Issue by IntegralWorks** [tensorflow/tensorflow#34216]
<em>Please make sure that this is a build/installation issue. As per our [GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md), we only address code/doc bugs, performance issues, feature requests and build/installation issues on GitHub. tag:build_template</em>

**System information**
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Linux Ubuntu 18.04
- TensorFlow installed from (source or binary): Docker, `docker run --gpus all -it --rm tensorflow/tensorflow:latest-gpu-py3`

- TensorFlow version: 2.0.0
- Python version: 3.6.8 
- Installed using virtualenv? pip? conda?: Docker
- Bazel version (if compiling from source): n/a
- GCC/Compiler version (if compiling from source): n/a
- CUDA/cuDNN version: nvcc: NVIDIA (R) Cuda compiler driver
Copyright (c) 2005-2018 NVIDIA Corporation
Built on Sat_Aug_25_21:08:01_CDT_2018
Cuda compilation tools, release 10.0, V10.0.130

cudNN: does not seem to come with Docker image. I am trying to use

`sudo docker cp libcudnn7_7.6.5.32-1+cuda10.0_amd64.deb [container ID]:/ && sudo docker cp libcudnn7-dev_7.6.5.32-1+cuda10.0_amd64.deb [container ID]:/ && sudo docker cp libcudnn7-doc_7.6.5.32-1+cuda10.0_amd64.deb [container ID]:/` on the host then

`dpkg -i libcudnn7_7.6.5.32-1+cuda10.0_amd64.deb && dpkg -i libcudnn7-dev_7.6.5.32-1+cuda10.0_amd64.deb && dpkg -i libcudnn7-doc_7.6.5.32-1+cuda10.0_amd64.deb`

- GPU model and memory: RTX 2060 6GB

**Describe the problem**

If I don't install cuDNN manually, dlib compiles "successfully" but will not use CUDA. If I install cudNN via dpkg, I get this error during the build:

-- Using CMake version: 3.10.2
-- Compiling dlib version: 19.18.99
-- Found system copy of libpng: /usr/lib/x86_64-linux-gnu/libpng.so;/usr/lib/x86_64-linux-gnu/libz.so
-- Found system copy of libjpeg: /usr/lib/x86_64-linux-gnu/libjpeg.so
-- Looking for cuDNN install...
-- Found cuDNN: /usr/lib/x86_64-linux-gnu/libcudnn.so
-- Enabling CUDA support for dlib.  DLIB WILL USE CUDA
-- C++11 activated.
CMake Error: The following variables are used in this project, but they are set to NOTFOUND.
Please set them or make sure they are set and tested correctly in the CMake files:
CUDA_cublas_LIBRARY (ADVANCED)
    linked by target "dlib" in directory /home/project/dlib/dlib
CUDA_curand_LIBRARY (ADVANCED)
    linked by target "dlib" in directory /home/project/dlib/dlib
CUDA_cusolver_LIBRARY (ADVANCED)
    linked by target "dlib" in directory /home/project/dlib/dlib

-- Configuring incomplete, errors occurred!
See also "/home/project/dlib/build/CMakeFiles/CMakeOutput.log".
See also "/home/project/dlib/build/CMakeFiles/CMakeError.log".
root@c7f1e218bc6f:/home/project/dlib/build# 

I also tried

`sudo docker cp cudnn-10.0-linux-x64-v7.6.5.32.tgz [c7f1e218bc6f]:/ `

`tar -xzvf cudnn-10.0-linux-x64-v7.6.5.32.tgz && cp cuda/include/cudnn.h /usr/local/cuda/include && cp cuda/lib64/libcudnn* /usr/local/cuda/lib64 && chmod a+r /usr/local/cuda/include/cudnn.h /usr/local/cuda/lib64/libcudnn*`

`export LD_LIBRARY_PATH=$LD_LIBRARY_PATH:/usr/local/cuda
export LD_LIBRARY_PATH=$LD_LIBRARY_PATH:/usr/local/cuda/include
export LD_LIBRARY_PATH=$LD_LIBRARY_PATH:/usr/local/cuda/lib64`

**Provide the exact sequence of commands / steps that you executed before running into the problem**

`sudo docker run --gpus all -it --rm tensorflow/tensorflow:latest-gpu-py3 #run container`

`python -c "import tensorflow as tf; print(tf.reduce_sum(tf.random.normal([1000, 1000])))" #hello world`
note: this^ runs just fine.

`sudo docker cp libcudnn7_7.6.5.32-1+cuda10.0_amd64.deb [container ID]:/ && sudo docker cp libcudnn7-dev_7.6.5.32-1+cuda10.0_amd64.deb [container ID]:/ && sudo docker cp libcudnn7-doc_7.6.5.32-1+cuda10.0_amd64.deb [container ID]:/`

`dpkg -i libcudnn7_7.6.5.32-1+cuda10.0_amd64.deb && dpkg -i libcudnn7-dev_7.6.5.32-1+cuda10.0_amd64.deb && dpkg -i libcudnn7-doc_7.6.5.32-1+cuda10.0_amd64.deb`

`apt update && apt upgrade && apt install git cmake libx11-dev nano
cd home && mkdir project && cd project 
git clone https://github.com/davisking/dlib.git #setup dlib`

`cd dlib && mkdir build; cd build; cmake ..; cmake --build .`

**Any other info / logs**
Include any logs or source code that would be helpful to diagnose the problem. If including tracebacks, please include the full traceback. Large logs and files should be attached.

CMakeOutput.log: https://pastebin.com/QC1kdejw
CMakeError.log: https://pastebin.com/JfQc02Qb

**Comment by ravikyram** [comment]
Can you try using bazel instead of cmake and let us know how it progresses. Thanks!

**Comment by ravikyram** [comment]
@IntegralWorks 

Any update on this issue please. Thanks!

**Comment by IntegralWorks** [comment]
@ravikyram
Sorry, new to GitHub and I'm busy this week with examinations. I hope this
message reaches you; I'm replying on email.
Unfortunately I decided using Docker for my project was not worth it and I
ended up successfully installing dlib with CUDA support on bare
metal/regular install.
I noticed that the key difference between my regular install and the Docker
container is that the container seems to refuse multiple versions of cuDNN.
If you're interested in how this issue turns out I highly suggest finding
someone who needs cuDNN in their project and wants to use the Tensorflow
GPU container.

Even stranger is that nvcc works out the box for the Tensorflow container,
whereas for a manual install the user-developer must meticulously follow
nVidia's documentation to the letter, such as adding things to .bashrc and
the such. But cuDNN seems broken even upon installation either through the
tar file or the deb package. With a properly, meticulously installed
regular CUDA installation cuDNN works seamlessly--the point I'm trying to
make is I might so bold as to suggest the Tensorflow GPU container might
have a semi-broken CUDA installation. Of course, it's more likely that I
just don't know how to install cuDNN on the Tensorflow GPU container
properly--but there doesn't seem to be any documentation(!) on this edge
case(?) I don't see how it could be an edge case if dlib isn't some obscure
library.

tl;dr
I have no idea how to use Bazel, I've been trained to use gcc, clang, cmake
etc. All compilation methods I know of cause dlib to *detect *CUDA, only to
say "cuDNN is broken, sorry". Therefore I want to emphasize that right now*
dlib does **not compile with GPU support on the Tensorflow GPU Docker
Container, *despite the fact that *intuitively* you'd expect it to just be
plug-and-play, at least after installing cuDNN via the *documented
instructions on nVidia's websites.*

**Comment by ravikyram** [comment]
@IntegralWorks 

CUDA and cuDNN can be installed by following these [instructions](https://www.tensorflow.org/install/gpu#install_cuda_with_apt). Thanks!

**Comment by ravikyram** [comment]
@IntegralWorks 

Any update on this issue please. Thanks!

**Comment by ravikyram** [comment]
Automatically closing due to lack of recent activity. Please update the issue when new information becomes available, and we will reopen the issue. Thanks!
